### PR TITLE
extension: define extension storage API w/ filesystem implementation

### PR DESCRIPTION
**Pull request by jazzdan**
This explicitly only stores the contents of the module's Tiltfile and doesn't deal with any of the other metadata yet, just to unblock work on other parts of functionality.

### CODE REVIEW

File: internal/tiltfile/extension/store.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		return "", nil
+	}
+
+	return "", err
+}
+
+func (s *LocalStore) Write(ctx context.Context, contents ModuleContents) (string, error) {
+	moduleDir := filepath.Join(s.baseDir, contents.Name)
+	if err := os.MkdirAll(moduleDir, os.FileMode(0700)); err != nil {
+		return "", fmt.Errorf("couldn't store module %s: %v", contents.Name, err)
```
Comment (landism) [inline internal/tiltfile/extension/store.go]: mind testing manually and/or in a unit test that the error message for this and below include the path that we failed to write?

it'd be super annoying if it were just like "couldn't store module openfaas: permission denied" and there was no pointer to what permissions needed to be fixed

also....errors.Wrapf!

File: internal/tiltfile/extension/store.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	tiltfilePath := filepath.Join(moduleDir, tiltfile.FileName)
+	// TODO(dmiller): store hash, source, time fetched
+	if err := ioutil.WriteFile(tiltfilePath, []byte(contents.TiltfileContents), os.FileMode(0700)); err != nil {
```
Comment (landism) [inline internal/tiltfile/extension/store.go]: this should probably be 0600

File: internal/tiltfile/extension/store.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	Stat(ctx context.Context, moduleName string) (string, error)
+	Write(ctx context.Context, contents ModuleContents) (string, error)
+}
+
+// TODO(dmiller): should this include an integrity hash?
+type ModuleContents struct {
+	Name             string
+	TiltfileContents string
+	GitCommitHash    string
+	Source           string
```
Comment (landism) [inline internal/tiltfile/extension/store.go]: ha, when I read "Source", I interpreted it as the source code of the extension, and was trying to figure out how it differed from `TiltfileContents`

maybe we want a `type ExtensionRegistry string`?

File: internal/tiltfile/extension/store.go (hunk 1 above)
Comment (jazzdan) [reply internal/tiltfile/extension/store.go]: I'll just manually include it.

**Pull request merged**